packages/@ember/-internals/routing/lib/system/route.ts

```diff
@@ -1507,6 +1507,17 @@ class Route extends EmberObject implements IRoute {
     @public
   */
   renderTemplate(_controller: any, _model: {}) {
+    if (this.renderTemplate !== Route.prototype.renderTemplate) {
+      deprecate('Usage of `renderTemplate` is deprecated.', false, {
+        id: 'route-render-template',
+        until: '4.0.0',
+        url: 'https://deprecations.emberjs.com/v3.x/#toc_route-render-template',
+        for: 'ember-source',
+        since: {
+          enabled: '3.27.0',
+        },
+      });
+    }
```

Comment:
You can invert the condition, and add it as the second arg to `deprecate`, something like this (not sure if prettier will be happy, but wanted to give an example):

```suggestion
    deprecate(
      'Usage of `renderTemplate` is deprecated.',
      this.renderTemplate === Route.prototype.renderTemplate,
      {
        id: 'route-render-template',
        until: '4.0.0',
        url: 'https://deprecations.emberjs.com/v3.x/#toc_route-render-template',
        for: 'ember-source',
        since: {
          enabled: '3.27.0',
        },
      }
    );
```

Reply:
Also, another idea here:

Should we issue the deprecation from within `Route#setup` instead of in the super class implementation of `renderTemplate`? I'm vaguely worried about folks not calling `this._super` and therefore never seeing the deprecation...

Reply:
Sure, I'll make both those changes 👍

Reply:
I made those changes and fixed up a bunch more failing tests which were caught by your suggestion to move to `Route#setup` instead